spec/jobs/irs_attempts_events_batch_job_spec.rb

```diff
@@ -62,87 +76,207 @@
+          expect(IrsAttemptsApi::EnvelopeEncryptor).to receive(:encrypt).with(
+            data: events.pluck(:jwe).join("\r\n"),
+            timestamp: start_time,
+            public_key_str: encoded_public_key,
+          )
 
-        expect_any_instance_of(described_class).to receive(:logger_info_hash).with(
-          name: 'IRSAttemptsEventJob',
-          start_time: Time.zone.now,
-          end_time: Time.zone.now,
-          duration_ms: 0.1234,
-          events_count: 2,
-          file_bytes_size: 19,
-        )
+          expect_any_instance_of(described_class).to receive(
+            :create_and_upload_to_attempts_s3_resource,
+          ).with(
+            bucket_name: bucket_name,
+            filename: envelope_encryptor_result[:filename],
+            encrypted_data: envelope_encryptor_result[:encrypted_data],
+          )
 
-        result = IrsAttemptsEventsBatchJob.perform_now(start_time)
+          expect_any_instance_of(described_class).to receive(:logger_info_hash).with(
+            name: 'IRSAttemptsEventJob',
+            start_time: Time.zone.now,
+            end_time: Time.zone.now,
+            duration_ms: 0.1234,
+            events_count: 2,
+            file_bytes_size: 19,
+          )
 
-        expect(result).not_to be_nil
-        expect(result[:filename]).to eq(
-          envelope_encryptor_result[:filename],
-        )
+          result = IrsAttemptsEventsBatchJob.perform_now(start_time)
 
-        expect(result[:iv]).to eq(
-          Base64.strict_encode64(envelope_encryptor_result[:iv]),
-        )
+          expect(result).not_to be_nil
+          expect(result[:filename]).to eq(
+            envelope_encryptor_result[:filename],
+          )
 
-        expect(result[:encrypted_key]).to eq(
-          Base64.strict_encode64(envelope_encryptor_result[:encrypted_key]),
-        )
+          expect(result[:iv]).to eq(
+            Base64.strict_encode64(envelope_encryptor_result[:iv]),
+          )
 
-        expect(result[:requested_time]).to eq(
-          IrsAttemptsApi::EnvelopeEncryptor.formatted_timestamp(start_time),
-        )
-      end
+          expect(result[:encrypted_key]).to eq(
+            Base64.strict_encode64(envelope_encryptor_result[:encrypted_key]),
+          )
 
-      context 'When irs delete events feature flag and s3 put_object response is true' do
-        before do
-          allow(IdentityConfig.store).to receive(:irs_attempt_api_delete_events_after_s3_upload).
-            and_return(true)
+          expect(result[:requested_time]).to eq(
+            IrsAttemptsApi::EnvelopeEncryptor.formatted_timestamp(start_time),
+          )
         end
 
-        it 'delete the events from redis' do
-          IrsAttemptsEventsBatchJob.perform_now
-          events = IrsAttemptsApi::RedisClient.new.read_events(timestamp: start_time)
-          expect(events.count).to eq 0
+        context 'When irs delete events feature flag and s3 put_object response is true' do
```

Comment:
```suggestion
        context 'When irs delete_events feature flag and s3 put_object response are true' do
```

Super nitty, and this isn't code you added, but my brain spent a bit trying to figure out what it means for the IRS to delete events.